devicemodel/hw/pci/virtio/virtio_block.c

```diff
@@ -487,11 +546,13 @@ virtio_blk_deinit(struct vmctx *ctx, struct pci_vdev *dev, char *opts)
 	if (dev->arg) {
 		DPRINTF(("virtio_blk: deinit\n"));
 		blk = (struct virtio_blk *) dev->arg;
-		bctxt = blk->bc;
-		if (blockif_flush_all(bctxt))
-			WPRINTF(("vrito_blk:"
-				"Failed to flush before close\n"));
-		blockif_close(bctxt);
+		/* De-init virtio-blk device only on valid bctxt*/
+		if (!blk->dummy_bctxt) {
+			bctxt = blk->bc;
+			if (blockif_flush_all(bctxt))
+				WPRINTF(("vrito_blk: Failed to flush before close\n"));
```

Comment:
```suggestion
				WPRINTF(("virtio_blk: Failed to flush before close\n"));
```

Reply:
working on a clean-up code patch and will fix this along with that patch.